schema.py

```diff
@@ -0,0 +1,115 @@
+"""Code generation script.
+
+Builds the schema from the public endpoint
+based on the OpenAPI specification.
+"""
+
+import os
+from subprocess import Popen
+
+import requests
+from pydantic import Field
+from pydantic_core import Url
+from pydantic_settings import BaseSettings
+
+
+class Settings(BaseSettings):
+    """Settings for the code generation.
+
+    Each setting can be overridden by an environment variable.
+
+    Parameters
+    ----------
+    out_dir : str
+        The directory where the generated schema will be saved.
+
+    module_name : str
+        The name of the module to be generated.
+
+    openapi_source : str
+        The source of the OpenAPI schema.
+
+    schema_name : str
+        The name of the schema file to be generated.
+
+    delete_after : bool
+        Whether to delete the schema file after the code generation.
+
+    """
+
+    out_dir: str = Field("qcog_python_client/schema/generated_schema/", alias="OUT_DIR")
+    module_name: str = Field("models", alias="MODULE_NAME")
+    openapi_source: Url = Field(
+        "https://dev.qognitive.io/openapi.json", alias="OPENAPI_SOURCE"
+    )
+    schema_name: str = Field("schema", alias="SCHEMA_NAME")
+    delete_after: bool = Field(True, alias="DELETE_AFTER")
+
+
+def pull_openapi_schema(from_source: str, schemafile_name: str) -> str:
+    """Pull the OpenAPI schema from the source.
+
+    Generate a new schema.json file from the source at the given address.
+    """
+    response = requests.get(from_source)
```

Comment:
might be useful to provide `timeout` arg to fail after several seconds of wait time instead of infinite wait in case our backend is down